False positive missing-param-doc on numpy style when ` : <type>` omitted

### Bug description

```python
"""A module."""


# https://numpydoc.readthedocs.io/en/latest/format.html#parameters
# The numpy style guide describes that parameters can be defined without
# being followed with a semi-colon and the type, i.e. arg2 below should
# not fail either of the raised checks.

def func(arg1: bool, arg2: bool):
    """Return args.

    Parameters
    ----------
    arg1 : bool
        arg1

    arg2
        arg2
    """
    return arg1, arg2
```


### Configuration

```ini
[MASTER]
load-plugins=pylint.extensions.docparams

[PARAMETER_DOCUMENTATION]
default-docstring-type=numpy
```


### Command used

```shell
pylint pylint_bug.py
```


### Pylint output

```shell
************* Module pylint_bug
pylint_bug.py:9:0: W9015: "arg2" missing in parameter documentation (missing-param-doc)
pylint_bug.py:9:0: W9012: Missing return type documentation (missing-return-type-doc)
```


### Expected behavior

Would not have expected either `missing-param-doc` or `missing-return-type-doc` checks to have failed.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.4
Python 3.8.2 (tags/v3.8.2:7b3ab59, Feb 25 2020, 23:03:10) [MSC v.1916 64 bit (AMD64)]
```


### OS / Environment

VS Code 1.61.2
Windows 10 Pro 21H1

### Additional dependencies

_No response_

--- a/ChangeLog
+++ b/ChangeLog
@@ -103,6 +103,11 @@ Release date: TBA
 
   Closes #3733
 
+* ``mising-param-doc`` now correctly handles Numpy parameter documentation without
+  explicit typing
+
+  Closes #5222
+
 * Update ``literal-comparison``` checker to ignore tuple literals
 
   Closes #3031

--- a/doc/whatsnew/2.12.rst
+++ b/doc/whatsnew/2.12.rst
@@ -123,3 +123,8 @@ Other Changes
   keyword parameters
 
   Closes #3733
+
+* ``mising-param-doc`` now correctly handles Numpy parameter documentation without
+  explicit typing
+
+  Closes #5222

--- a/pylint/extensions/_check_docs_utils.py
+++ b/pylint/extensions/_check_docs_utils.py
@@ -785,7 +785,7 @@ class NumpyDocstring(GoogleDocstring):
                 continue
 
             # check if parameter has description only
-            re_only_desc = re.match(r"\\s*  (\\*{0,2}\\w+)\\s*:?\n", entry)
+            re_only_desc = re.match(r"\s*  (\*{0,2}\w+)\s*:?\n", entry)
             if re_only_desc:
                 param_name = match.group(1)
                 param_desc = match.group(2)

--- a/pylint/extensions/_check_docs_utils.py
+++ b/pylint/extensions/_check_docs_utils.py
@@ -23,7 +23,7 @@
 """Utility methods for docstring checking."""
 
 import re
-from typing import List
+from typing import List, Set, Tuple
 
 import astroid
 from astroid import nodes
@@ -731,9 +731,8 @@ class NumpyDocstring(GoogleDocstring):
 
     re_param_line = re.compile(
         fr"""
-        \s*  (\*{{0,2}}\w+)                                                 # identifier with potential asterisks
-        \s*  :
-        \s*  (?:({GoogleDocstring.re_multiple_type})(?:,\s+optional)?)?     # optional type declaration
+        \s*  (\*{{0,2}}\w+)(\s?(:|\n))                                      # identifier with potential asterisks
+        \s*  (?:({GoogleDocstring.re_multiple_type})(?:,\s+optional)?\n)?   # optional type declaration
         \s* (.*)                                                            # optional description
     """,
         re.X | re.S,
@@ -771,6 +770,38 @@ class NumpyDocstring(GoogleDocstring):
     re_yields_line = re_returns_line
 
     supports_yields = True
+
+    def match_param_docs(self) -> Tuple[Set[str], Set[str]]:
+        """Matches parameter documentation section to parameter documentation rules"""
+        params_with_doc = set()
+        params_with_type = set()
+
+        entries = self._parse_section(self.re_param_section)
+        print(entries)
+        entries.extend(self._parse_section(self.re_keyword_param_section))
+        for entry in entries:
+            match = self.re_param_line.match(entry)
+            if not match:
+                continue
+
+            # check if parameter has description only
+            re_only_desc = re.match(r"\\s*  (\\*{0,2}\\w+)\\s*:?\n", entry)
+            if re_only_desc:
+                param_name = match.group(1)
+                param_desc = match.group(2)
+                param_type = None
+            else:
+                param_name = match.group(1)
+                param_type = match.group(3)
+                param_desc = match.group(4)
+
+            if param_type:
+                params_with_type.add(param_name)
+
+            if param_desc:
+                params_with_doc.add(param_name)
+
+        return params_with_doc, params_with_type
 
     @staticmethod
     def min_section_indent(section_match):

--- a/tests/extensions/test_check_docs.py
+++ b/tests/extensions/test_check_docs.py
@@ -292,35 +292,6 @@ class TestParamDocChecker(CheckerTestCase):
         ):
             self.checker.visit_functiondef(node)
 
-    @set_config(default_docstring_type="numpy")
-    def test_no_type_func_params_in_numpy_docstring(self) -> None:
-        """Example of a function with NumPy style parameter without type annotation
-        documentation in the docstring
-        """
-        node = astroid.extract_node(
-            """
-        def func(arg1: bool, arg2: bool):
-            '''Return args.
-
-            Parameters
-            ----------
-            arg1 : bool
-                arg1
-
-            arg2
-                arg2
-
-            Returns
-            ----------
-            bool
-            bool
-            '''
-            return arg1, arg2
-        """
-        )
-        with self.assertNoMessages():
-            self.checker.visit_functiondef(node)
-
     @set_config(accept_no_param_doc=True)
     def test_tolerate_no_param_documentation_at_all(self) -> None:
         """Example of a function with no parameter documentation at all
@@ -1367,13 +1338,17 @@ class TestParamDocChecker(CheckerTestCase):
     def test_finds_args_without_type_numpy(self) -> None:
         node = astroid.extract_node(
             '''
-        def my_func(named_arg, *args):
+        def my_func(named_arg, typed_arg: bool, untyped_arg, *args):
             """The docstring
 
             Args
             ----
             named_arg : object
                 Returned
+            typed_arg
+                Other argument without numpy type annotation
+            untyped_arg
+                Other argument without any type annotation
             *args :
                 Optional Arguments
 
@@ -1386,7 +1361,9 @@ class TestParamDocChecker(CheckerTestCase):
                 return named_arg
         '''
         )
-        with self.assertNoMessages():
+        with self.assertAddsMessages(
+            MessageTest(msg_id="missing-type-doc", node=node, args=("untyped_arg",))
+        ):
             self.checker.visit_functiondef(node)
 
     def test_finds_args_with_xref_type_google(self) -> None:

--- a/tests/extensions/test_check_docs.py
+++ b/tests/extensions/test_check_docs.py
@@ -292,6 +292,35 @@ class TestParamDocChecker(CheckerTestCase):
         ):
             self.checker.visit_functiondef(node)
 
+    @set_config(default_docstring_type="numpy")
+    def test_no_type_func_params_in_numpy_docstring(self) -> None:
+        """Example of a function with NumPy style parameter without type annotation
+        documentation in the docstring
+        """
+        node = astroid.extract_node(
+            """
+        def func(arg1: bool, arg2: bool):
+            '''Return args.
+
+            Parameters
+            ----------
+            arg1 : bool
+                arg1
+
+            arg2
+                arg2
+
+            Returns
+            ----------
+            bool
+            bool
+            '''
+            return arg1, arg2
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(node)
+
     @set_config(accept_no_param_doc=True)
     def test_tolerate_no_param_documentation_at_all(self) -> None:
         """Example of a function with no parameter documentation at all

